packages/dds/tree/src/shared-tree/schematizingTreeView.ts

```diff
@@ -244,28 +245,38 @@ export class SchematizingSimpleTreeView<
 			| void,
 		params?: RunTransactionParams,
 	): TransactionResultExt<TSuccessValue, TFailureValue> | TransactionResult {
-		this.checkout.transaction.start();
-		const preconditions = params?.preconditions ?? [];
-		for (const constraint of preconditions) {
-			switch (constraint.type) {
-				case "nodeInDocument": {
-					const node = getOrCreateInnerNode(constraint.node);
-					const nodeStatus = getKernel(constraint.node).getStatus();
-					const kernel = getKernel(constraint.node);
-					assert(kernel !== undefined, 0x8c1 /* Node should have a kernel */);
-					if (nodeStatus !== TreeStatus.InDocument) {
-						throw new UsageError(
-							`Attempted to add a "nodeInDocument" constraint, but the node is not currently in the document. Node status: ${nodeStatus}`,
-						);
+		const validateConstraints = (
```

Comment:
Nit: since this function is also adding the constraints, I think "addConstraints" would make for a better name.